src/test/java/co/rsk/bitcoinj/script/P2shErpFederationRedeemScriptParserTest.java

```diff
@@ -327,6 +344,56 @@ public void isP2shErpFed_falseWithCustomRedeemScript() {
         Assert.assertFalse(P2shErpFederationRedeemScriptParser.isP2shErpFed(customRedeemScript.getChunks()));
     }
 
+    @Test
+    public void getMultiSigType_shouldReturnP2shErpFedType() {
+        Assert.assertEquals(MultiSigType.P2SH_ERP_FED, p2shErpFederationRedeemScriptParser.getMultiSigType());
+    }
+
+    @Test
+    public void getM_shouldReturnM() {
+        int expectedM = defaultRedeemScriptKeys.size() / 2 + 1;
+        Assert.assertEquals(expectedM, p2shErpFederationRedeemScriptParser.getM());
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyExists_shouldReturnIndex() {
+        for (int i = 0; i < defaultRedeemScriptKeys.size(); i++) {
+            BtcECKey expectedKey = defaultRedeemScriptKeys.get(i);
+            Assert.assertEquals(i, p2shErpFederationRedeemScriptParser.findKeyInRedeem(expectedKey));
+        }
+    }
+
+    @Test
+    public void findKeyInRedeem_whenKeyDoesNotExists_shouldThrowIllegalStateException() {
+        BtcECKey unknownKey = BtcECKey.fromPrivate(BigInteger.valueOf(1234567890L));
+        for (int i = 0; i < defaultRedeemScriptKeys.size(); i++) {
```

Comment:
What's the point of this iteration? Isn't it testing the same thing over and over?

Reply:
It is useless. Solved.